packages/core/src/services/learnAndEarn/list.ts

```diff
@@ -233,12 +233,44 @@ export async function listLessons(userId: number, levelId: number) {
             }
         );
 
+        const rewardAvailable = await getRewardAvailable(levelId);
+
         return {
             totalPoints,
+            rewardAvailable,
             completedToday: completedToday > 0,
             lessons: mappedLessons,
         };
     } catch (error) {
         throw new BaseError('LIST_LESSONS_FAILED', 'list lessons failed');
     }
 }
+
+const getRewardAvailable = async (levelId: number): Promise<number | null> => {
+    const level = await models.learnAndEarnLevel.findOne({
+        attributes: ['rewardLimit'],
+        where: {
+            id: levelId,
+        }
+    });
+
+    if (!level?.rewardLimit) {
+        return null;
+    }
+
+    const payments = await models.learnAndEarnPayment.findOne({
+        attributes: [
+            [fn('sum', col('amount')), 'amount'],
+        ],
+        where: {
+            levelId,
+            status: 'paid'
+        }
+    });
+
+    if (!payments?.amount) {
+        return level.rewardLimit
+    }
+
+    return level.rewardLimit - payments.amount;
```

Comment:
This is tricky to get to exactly zero. And the frontend only needs a boolean. So, I'd suggest, returning true only if the remaining amount is bigger than the max rewards a user can get. To avoid leaving tokens to disburse, what we can do is, when a user finishes a level, we make a calculation to see if the total to receive is less that the amount available, and if so, generate the signature.

This is still not ideal, but we can iterate later. And also, this can be improved with some [subgraph entities](https://github.com/impactMarket/subgraph/issues/81).